app/controllers/users_controller.rb

```diff
@@ -155,4 +156,12 @@ def login_user(user)
   def parameter_filter_context
     Foreman::Controller::Parameters::User::Context.new(:ui, controller_name, params[:action], editing_self?)
   end
+
+  def verify_active_session
+    if params[:status].blank? && session[:user].present? && User.exists?(session[:user])
+      warning _("You have already Logged-In.")
+      redirect_to root_path
```

Comment:
This should be the previous URL preferably, else `hosts_path` to match login behaviour, and to work for users without dashboard permissions.